extra_geom/detectors.py

```diff
@@ -854,6 +856,191 @@ def to_distortion_array(self, allow_negative_xy=False):
         return super().to_distortion_array(allow_negative_xy)
 
 
+class AGIPD_500K2GGeometry(DetectorGeometryBase):
+    """Detector layout for AGIPD-500k
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    You won't normally instantiate this class directly:
+    use one of the constructor class methods to create or load a geometry.
+    """
+    detector_type_name = 'AGIPD-500K2G'
+    pixel_size = 2e-4  # 2e-4 metres == 0.2 mm
+    frag_ss_pixels = 64
+    frag_fs_pixels = 128
+    expected_data_shape = (8, 512, 128)
+    n_modules = 8
+    n_tiles_per_module = 8
+
+    @classmethod
+    def default(cls, asic_gap=2, panel_gap=(16, 30), unit=pixel_size):
```

Comment:
Looking at this now, maybe it's clearer to have just `from_origin`, with a default so the `origin` parameter is optional. What do you think?

Reply:
Yep, that's good